src/scraping/collector.py

```
@@ -0,0 +1,123 @@
+        self.batch_size = SCRAPING_BATCH_SIZE
+        self.delay = SCRAPING_DELAY
+
+    async def __aenter__(self):
+        self.session = aiohttp.ClientSession(auth=self.auth)
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        if self.session:
+            await self.session.close()
+
+    async def collect_leads(self, criteria: Dict[str, Any]) -> List[Dict[str, Any]]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)